processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -216,18 +217,12 @@ public File mergeQueryableIndex(
       ProgressIndicator progress
   ) throws IOException
   {
+    List<IndexableAdapter> indexAdapteres = new ArrayList<IndexableAdapter>();
+    for (QueryableIndex index : indexes) {
+      indexAdapteres.add(new QueryableIndexIndexableAdapter(index));
+    }
     return merge(
-        Lists.transform(
```

Comment:
you could avoid the for loop by simply doing a `Lists.newArrayList(Iterables.transform(..))`

It would also be good to add a comment here that we are materializing the list for performance reasons.